docs(cache): added docs for cache options - build dependencies (#3398)
webpack/webpack.js.org: src/content/configuration/other-options.md

While setting `cache.type` to `filesystem` opens up more options for configuration.

### `cache.buildDependencies`

`Object: [path]`

> `cache.buildDependencies` is an object of arrays of additional code dependencies of the build. webpack will use an hash of these items and all dependencies to invalidate the filesystem cache.

Reviewer: ```suggestion
`cache.buildDependencies` is an object of arrays of additional code dependencies of the build. webpack will use a hash of these items and all dependencies to invalidate the filesystem cache.
```

Defaults to `webpack/lib` to get all dependencies of webpack.

> T> It's recommended to set `cache.buildDependencies.config: [__filename]` in `webpack.config.js` to get the config and all dependencies.

Reviewer 2: ```suggestion
T> It's recommended to set `cache.buildDependencies.config: [__filename]` in your webpack configuration to get the latest configuration and all dependencies.
```

```javascript
module.exports = {
  cache: {
    buildDependencies: {

>       // recommended to invalidate cache on config changes

Reviewer 2: Duplication of what you've said earlier in the W> block, maybe not necessary?

>       // This also makes all dependencies of this file build dependencies

Reviewer 2: ```suggestion
      // This also makes all dependencies of this file - build dependencies
```

};
```

### `cache.managedPaths`

`Array[path]`

> `cache.managedPaths` is an array of package-manager only managed paths. webpack will avoid hashing/timestamping them, assume the version is unique and will use this as snapshot. (for both memory and filesystem cache) Defaults to the `node_modules` webpack is inside of.

Reviewer 2: ```suggestion
`cache.managedPaths` is an array of package-manager only managed paths. webpack will avoid hashing and timestamping them, assume the version is unique and will use it as a snapshot (for both memory and filesystem cache).
```
